src/main/java/com/code_intelligence/jazzer/mutation/mutator/proto/TypeLibrary.java

```diff
@@ -69,9 +67,10 @@ private static <T extends Builder> AnnotatedType getBaseTypeWithPresence(
         return new TypeHolder<ByteString>() {}.annotatedType();
       case STRING:
         return new TypeHolder<String>() {}.annotatedType();
+      case ENUM:
+        return new TypeHolder<EnumValueDescriptor>() {}.annotatedType();
```

Comment:
Is this used? `EnumValueDescriptorMutator`s are generated dynamically.

Reply:
True, it wasn't used but still needed to be a valid type to add `@NotNull` to. I now refactored the change a bit more so that this is actually used. Most of the branches specific to enums went away, please take another look.

Reply:
Mapping `ENUM` to `EnumValueDescriptor`, together with the dynamic `MutatorFactory`, is really clean. I like it!